practice/rain.py

```diff
@@ -0,0 +1,193 @@
+        if year_in_date_data not in yearly_totals:
+            yearly_totals.update({year_in_date_data: 0})
+        yearly_totals[year_in_date_data] += amt
+    return yearly_totals
+
+
+def calc_daily_avg_stats(totals):
+    """Return a dictionary of days and average amounts.
+
+    >>> sorted(calc_daily_avg_stats({'30-MAR-2016': 1, '29-MAR-2016': 5,
+    ...                      '30-MAR-2015': 3, '29-MAR-2015': 3}).items())
+    [('29-MAR', 4.0), ('30-MAR', 2.0)]
+    """
+    daily_sums = {}
+    for date_data, amt in totals.items():
+        day_in_date_data = '-'.join(date_data.split('-')[0:2])
+        if day_in_date_data not in daily_sums:
+            daily_sums.update({day_in_date_data: [float(amt)]})
+        else:
+            daily_sums[day_in_date_data].append(amt)
+    daily_averages = {k: statistics.mean(v) for k, v in daily_sums.items()}
+    return daily_averages
+
+
+def get_inquiry_date():
+    """Return the user's selection of a date for subsequent inquiry."""
+    return input('What date would you like to check (DD-MMM): ').upper()
+
+
+def output_most_rain_stats(day, day_amt, year, year_amt):
+    """Print the wettest day and year statistics.
+
+    >>> output_most_rain_stats('29-MAR-2015', 25, '2012', 5520)
+    Wettest day was 29-MAR-2015, amount was 0.25"
+    Wettest year was 2012, amount was 55.2"
+    """
+    print('Wettest day was {}, amount was {}"'.format(day, day_amt/100))
+    print('Wettest year was {}, amount was {}"'.format(year, year_amt/100))
+
+
+def output_wettest_doy_stats(day, amt):
+    """Print the wettest day of the year amount
+
+    >>> output_wettest_doy_stats('29-MAR', 25)
+    Wettest day of the year is 29-MAR with 0.25"
+    """
+    print('Wettest day of the year is {} with {:.2f}"'.format(day, amt/100))
+
+
+def output_inquiry_stats(date, amt):
+    """Print the average amount for a requested day of year
+
+    >>> output_inquiry_stats('29-MAR', 25)
+    Average rainfall on 29-MAR is 0.25"
+    """
+    print('Average rainfall on {} is {:.2f}"'.format(date, amt/100))
+
+
+def read_store_html_source(site_name, file_name):
+    """Read and store local copy of html source, if the copy does not exist."""
+    if not os.path.isfile(file_name):
+        print('Loading copy of {}...'.format(file_name))
+        with urllib.request.urlopen(site_name) as rain_file:
+            web_text = rain_file.read().decode('utf-8')
+        with open(file_name, "w+") as file:
+            file.writelines(web_text)
+
+
+def read_html_file(filename):
+    """Read and return a list of the names and files of the rain gages."""
+    with open(filename) as file:
+        file_contents = file.read()
+    names = re.findall(r"<td>.* Rain Gage", file_contents)
+    files = re.findall(r"[a-z,_]*\.rain\">", file_contents)
+    name_list = [string[4:-10] for string in names]
+    file_list = [string[0:-2] for string in files]
+    name_file_list = zip(name_list, file_list)
+    return list(name_file_list)
```

Comment:
Same suggestion as above: split file reading from processing. Then you can test.